django/core/checks/messages.py

```diff
@@ -74,3 +87,16 @@ def __init__(self, *args, **kwargs):
 class Critical(CheckMessage):
     def __init__(self, *args, **kwargs):
         super().__init__(CRITICAL, *args, **kwargs)
+
+
+silenced_object_checks = defaultdict(WeakValueDictionary)
```

Comment:
Using a weak dictionary so that silencing a check for a temporary object (e.g. a temporary model or field in a test) doesn't cause it to have a permanent reference that keeps it sticking around forever.

Using a `defaultdict(WeakValueDictionary)` rather than `WeakKeyDictinoary` in order to support non-hashable objects (or objects whose hash *changes*... I found out that model `Field` does this, which is naughty, will file a ticket).

There's a proposal to add such a version of `WeakKeyDictionary` that similarly uses `id(obj)`, but no movement: https://bugs.python.org/issue44140